conda_forge_tick/recipe_parser/_parser.py

```diff
@@ -0,0 +1,394 @@
+            isinstance(n, jinja2.nodes.Assign) and
+            isinstance(n.node, jinja2.nodes.Const)
+        ):
+            if _config_has_key_with_selectors(jinja2_vals, n.target.name):
+                # selectors!
+
+                # this block runs if we see the key for the
+                # first time
+                if n.target.name in jinja2_vals:
+                    # we need to adjust the previous key
+                    # first get the data right after the key we have
+                    jinja2_data = (
+                        all_nodes[jinja2_vals[n.target.name][1]+1]
+                        .nodes[0]
+                        .data
+                    )
+
+                    # now pull out the selector and reset the key
+                    selector_re = SELECTOR_RE.match(jinja2_data)
+                    if selector_re is not None:
+                        selector = selector_re.group(1)
+                        new_key = n.target.name + CONDA_SELECTOR + selector
+                        jinja2_vals[new_key] = jinja2_vals[n.target.name]
+                        del jinja2_vals[n.target.name]
+                    else:
+                        assert False, jinja2_data
+
+                # now insert this key - selector is the next thing
+                jinja2_data = all_nodes[i+1].nodes[0].data
+                selector_re = SELECTOR_RE.match(jinja2_data)
+                if selector_re is not None:
+                    selector = selector_re.group(1)
+                    new_key = n.target.name + CONDA_SELECTOR + selector
+                    jinja2_vals[new_key] = (n.node.value, i)
+                else:
+                    assert False, jinja2_data
+            else:
+                jinja2_vals[n.target.name] = (n.node.value, i)
+
+    # we don't need the indexes into the jinja2 node list anymore
+    for key, val in jinja2_vals.items():
+        jinja2_vals[key] = jinja2_vals[key][0]
+
+    return jinja2_vals
+
+
+def _munge_line(line: str) -> str:
+    """turn lines like
+
+        key: val  # [sel]
+
+    to
+
+        key__###conda-selector###__sel: val
+    """
+    m = SPC_KEY_VAL_SELECTOR_RE.match(line)
+    if m:
+        spc, key, val, selector = m.group(1, 2, 3, 4)
+        new_key = key + CONDA_SELECTOR + selector
+        return spc + new_key + ':' + val + '\n'
+    else:
+        return line
+
+
+def _unmunge_line(line: str) -> str:
+    """turn lines like
+
+        key__###conda-selector###__sel: val
+
+    to
+
+        key: val  # [sel]
+
+    This one does the opposite of _munge_line above.
+    """
+    m = MUNGED_LINE_RE.match(line)
+    if m:
+        spc, key, selector, val = m.group(1, 2, 3, 4)
+        return spc + key + ': ' + val.strip() + '  # [' + selector + ']\n'
+    else:
+        return line
+
+
+def _demunge_jinja2_vars(meta: Union[dict, list]) -> Union[dict, list]:
+    """recursively iterate through dictionary / list and replace any instance
+    in any string of `<{` with '{{'
```

Comment:
could this use something more obscure than `<` ? this misbehaves in recipes that happen to have the string `<{{` for example in version constraints: https://github.com/conda-forge/boto3-feedstock/blob/1ff007ef13ef567734b52155c96f5f759f3bdd9e/recipe/meta.yaml#L31

Reply:
This is done by rumael.jinaj2, not this code. So you'll have to file an issue there.

Reply:
Can we workaround the `ruamel.jinja2` issue?

Reply:
See https://github.com/regro/cf-scripts/pull/850 and thanks for noticing this! 

I was too quick to say it could not be fixed here. My apologies for that!